src/main.rs

```diff
@@ -0,0 +1,275 @@
+                vertice_separation: 0.,
+                ..Default::default()
+            },
+            ..Default::default()
+        })
+        // The player is also a dynamic body with rotations locked
+        .insert(RigidBody::Dynamic)
+        .insert(RotationConstraints::lock())
+        // Disable friction and bounciness
+        .insert(PhysicMaterial {
+            friction: 0.,
+            restitution: 0.,
+            ..Default::default()
+        })
+        // Set the player speed to 0 initially
+        .insert(Velocity::from_linear(Vec3::default()))
+        .insert(Player {speed: 75.})
+        .insert(Health {value: 10});
+    
+}
+
+fn move_player(keyboard_input: Res<Input<KeyCode>>, mut query: Query<&mut Velocity, With<Player>>, player_q: Query<&Player>,
+    mut cam_query: Query<&mut Transform, With<Camera>>)  {
+    for mut velocity in query.iter_mut() {
+
+        let mut direction = Vec3::new(0., 0., 0.);
+
+        if keyboard_input.pressed(KeyCode::Left) {
+            direction += Vec3::new(-1.0, 0., 0.);
+        }
+
+        if keyboard_input.pressed(KeyCode::Right) {
+            direction += Vec3::new(1.0, 0., 0.);
+        }
+
+        if keyboard_input.pressed(KeyCode::Up) {
+            direction += Vec3::new(0., -1.0, 0.);
+        }
+
+        if keyboard_input.pressed(KeyCode::Down) {
+            direction += Vec3::new(0., 1.0, 0.);
+        }
+
+        direction = direction.normalize_or_zero();
+
+        for player in player_q.iter() {
+            *velocity = Velocity::from_linear(direction * player.speed);       
+            for mut transform in cam_query.iter_mut() {
+                transform.translation += direction * player.speed * 0.01 * 1.5;
+            }
+        }
+
+    }
+}
+
+fn move_camera(
+    keyboard_input: Res<Input<KeyCode>>,
+    mut query: Query<&mut Transform, With<Camera>>,
+) {
+    for mut transform in query.iter_mut() {
+        const SPEED: f32 = 1.;
+
+        let mut direction = Vec3::new(0., 0., 0.);
+
+        if keyboard_input.pressed(KeyCode::A) {
+            direction += Vec3::new(-SPEED, 0., 0.);
+        }
+
+        if keyboard_input.pressed(KeyCode::D) {
+            direction += Vec3::new(SPEED, 0., 0.);
+        }
+
+        if keyboard_input.pressed(KeyCode::W) {
+            direction += Vec3::new(0., -SPEED, 0.);
+        }
+
+        if keyboard_input.pressed(KeyCode::S) {
+            direction += Vec3::new(0., SPEED, 0.);
+        }
+
+        transform.translation += direction;
+    }
+}
+
+fn detect_collisions(mut events: EventReader<CollisionEvent>, mut health_query: Query<&mut Health, With<Player>>,
+    spike_query: Query<&Spike>, player_query: Query<&Player>) {
+
+    for event in events.iter() {
+        for mut health in health_query.iter_mut() {
+            for player in player_query.iter() {
+                match event {
+                    CollisionEvent::Started(spike_query, player) => {
```

Comment:
This is wrong usage of this API. calling match event {
CollisitonEvent::Started(data1, data2) will give you 2 datas which you need to check if they are Player and Spike, what you have done is just making a new variables spike_query and player which are shadowing you previous ones